### PR TITLE
New utility for making multiple changes to state at once (no JIRA)

### PR DESCRIPTION
**Pull request by MattPutnam**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs [hunk 1]
```diff
@@ -306,10 +306,10 @@
          :on-done (fn [{:keys [success? get-parsed-response]}]
                     (swap! state dissoc :blocker :editing?)
                     (if success?
-                      (do (swap! state dissoc :redacted?)
-                          ((:on-rename props) name)
+                      (do ((:on-rename props) name)
                           ((@refs "methodDetailsViewer") :clear-redacted-snapshot)
-                          (swap! state assoc :loaded-config (get-parsed-response) :blocker nil))
+                          (utils/multi-swap! state (assoc :loaded-config (get-parsed-response))
+                                                   (dissoc :redacted?)))
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs:312]: do you also need to `dissoc :blocker` here?

File: src/cljs/main/broadfcui/page/workspace/method_configs/publish.cljs [hunk 2]
```diff
@@ -38,9 +38,9 @@
          {:text "Publish"
           :onClick
           #(let [[ns n & fails] (input/get-and-validate refs "mcNamespace" "mcName")]
-             (swap! state assoc :validation-errors fails :error nil)
+             (utils/multi-swap! state (assoc :validation-errors fails) (dissoc :error))
              (when-not fails
-               (swap! state assoc :publishing? true :error nil)
+               (swap! state assoc :publishing? true)
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/workspace/method_configs/publish.cljs:43]: need `dissoc :error`?

**Review by davidangb — APPROVED** [review]
two comments inline, looks good otherwise. Did not test. :+1:

File: src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs (hunk 1 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs:312]: Nope, that's already getting `dissoc`ed above.

File: src/cljs/main/broadfcui/page/workspace/method_configs/publish.cljs (hunk 2 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/page/workspace/method_configs/publish.cljs:43]: Ditto here.

**Review by zarsky-broad — APPROVED** [review]
Okay, I've thought about this some more, and I've come around. Let's do it.

**Comment by zarsky-broad** [comment]
jenkins retest

**Pull request merged**